parl/common/replay_buffer.py

```diff
@@ -0,0 +1,115 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import random
+from error_handling import LastElementError
+
+
+class Experience(object):
+    def __init__(self, obs, action, reward, episode_end):
+        self.obs = obs
+        self.reward = reward
+        self.action = action
+        self.episode_end = episode_end
+
+
+class Sample(object):
+    def __init__(self, i, n):
```

Comment:
Comments on the two fields
i: starting index of the first exp in a sequence
n: length of the exp sequence